onnxruntime/core/providers/rocm/tunable/rocm_tunable.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <hip/hip_runtime.h>
+#include <hip/hip_fp16.h>
+
+// solving ODR violation due to provider_api.h
```

Comment:
I'm curious about what ODR violation this fixes.

Reply:
You better play around with `provider_api.h` a little bit to gain the sensation of the pain. It has some (re-)definition in that file. If you include the definition header and `provider_api.h` in the same time. And if you read it carefully, then you should have notice a special guard https://github.com/microsoft/onnxruntime/blob/454ea14c786fa521bafe7d174b5f211d1bf54dec/onnxruntime/core/framework/tunable.h#L22
comment it out, then you will have an interesting problem to play around ;) (or agony, warned)